### PR TITLE
Volume expansion support

### PR DESCRIPTION
**Pull request by Ashughorla**
# Proposed Changes

- Add volume expansion support to `storageClass`
- Add `csi-resizer` components
- Add tests

Fixes https://github.com/onmetal/gardener-extension-provider-onmetal/issues/179

### CODE REVIEW

File: pkg/apis/onmetal/types_cloudprofile.go [hunk 1]
```diff
@@ -48,8 +48,21 @@ type StorageClass struct {
 	Name string
 	// Type is referring to the VolumeClass to use for this StorageClass
 	Type string
+	// ResizePolicy describes the supported expansion policy of a VolumeClass.
+	// If not set default to Static expansion policy.
+	ResizePolicy ResizePolicy
```
Comment (ushabelgur) [inline pkg/apis/onmetal/types_cloudprofile.go:53]: @afritzler  we already have Type(Which is nothing but volumeClass) field in storage class and volume class already has the ResizePolicy field which decides if it's expandOnly or static, so adding new ResizePolicy field here seems redundant.

Comment (ushabelgur) [reply pkg/apis/onmetal/types_cloudprofile.go]: Changed logic to read resize policy from VolumeClass and set expandable property in StorageClass accordingly

File: pkg/controller/controlplane/valuesprovider_test.go [hunk 2]
```diff
@@ -165,6 +200,9 @@ var _ = Describe("Valueprovider Reconcile", func() {
 			Expect(err).NotTo(HaveOccurred())
 
 			cluster := &controller.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ns.Name,
+				},
```
Comment (afritzler) [inline pkg/controller/controlplane/valuesprovider_test.go:205]: No need to set the name here as the cluster resource is not applied.

File: pkg/controller/controlplane/valuesprovider_test.go [hunk 3]
```diff
@@ -214,6 +255,9 @@ var _ = Describe("Valueprovider Reconcile", func() {
...
```
Comment (afritzler) [inline pkg/controller/controlplane/valuesprovider_test.go:260]: No need to set the name here as the cluster resource is not applied.

File: pkg/controller/controlplane/valuesprovider_test.go [hunk 4]
```diff
@@ -120,12 +123,44 @@ var _ = Describe("Valueprovider Reconcile", func() {
...
+		})
 		It("should return an empty config chart value map if not storageclasses are present in the cloudprofile", func() {
 			providerCloudProfile := &apisonmetal.CloudProfileConfig{}
 			providerCloudProfileJson, err := json.Marshal(providerCloudProfile)
 			Expect(err).NotTo(HaveOccurred())
 
 			cluster := &controller.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ns.Name,
+				},
```
Comment (afritzler) [inline pkg/controller/controlplane/valuesprovider_test.go:163]: No need to set the name here as the cluster resource is not applied.

**Review by afritzler — CHANGES_REQUESTED** [review]

File: pkg/controller/controlplane/valuesprovider_test.go (hunk 2 above)
Comment (ushabelgur) [reply pkg/controller/controlplane/valuesprovider_test.go:205]: I am using to get onmetal client in valuesprovider.go

onmetalClient, namespace, err := onmetal.GetOnmetalClientAndNamespaceFromCloudProviderSecret(ctx, vp.Client(), **cluster.ObjectMeta.Name**)

**Review by afritzler — APPROVED** [review]

**Pull request merged**